connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -282,11 +337,14 @@ public boolean startTask(
             return false;
         }
 
+        WorkerTask existing = tasks.put(id, workerTask);
```

Comment:
Would `putIfAbsent` be a better option? Otherwise we'll replace the old entry which doesn't seem like what we want.